pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -339,17 +339,16 @@ func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) (ctrl.Result, erro
 		}
 		isvc.Status.PropagateStatus(v1beta1.PredictorComponent, status)
 	}
-	statusSpec, _ := isvc.Status.Components[v1beta1.PredictorComponent]
+	statusSpec := isvc.Status.Components[v1beta1.PredictorComponent]
 	if rawDeployment {
 		podLabelValue = constants.GetRawServiceLabel(predictorName)
 	} else {
 		podLabelValue = statusSpec.LatestCreatedRevision
 	}
-	podList, err := isvcutils.ListPodsByLabel(p.client, isvc.ObjectMeta.Namespace, podLabelKey, podLabelValue)
+	predictorPod, err := isvcutils.ListPodsByLabel(p.client, isvc.ObjectMeta.Namespace, podLabelKey, podLabelValue)
```

Comment:
should be `predictorPods` ? it is the pod list for the latest revision

Reply:
It actually only gets 1 pod for the latest revision: https://github.com/kserve/kserve/blob/master/pkg/apis/serving/v1beta1/inference_service_status.go#L486.

Reply:
though here is still returning a list of pods and in status propagation method it only fetches the first pod.